src/unnamed_rcv_thing/cleaning.py

```diff
@@ -0,0 +1,19 @@
+from unnamed_rcv_thing.profile import PreferenceProfile
+from unnamed_rcv_thing.ballot import Ballot
+from copy import deepcopy
+
+
+def remove_empty_ballots(pp: PreferenceProfile, keep_candidates: bool = False):
+    """
+    Returns a preference profile which is the input pp without empty ballots.
+    keep_candidates: use old set of candidates, even if some no longer appear
+    """
+    ballots_nonempty = [deepcopy(ballot) for ballot in pp.get_ballots() if ballot.ranking]
```

Comment:
immutability might let you avoid the somewhat expensive deepcopy here, was this in response to an issue? or as a precaution